### DISCUSSION TITLE
Define casing of column names

### DISCUSSION THREAD

**Discussion by Purtl** [laravel/framework#45011] [Q&A]
Heyho,
I have a project that has a db with columns named in camelCase. Probably not the most common thing, but it's consistent and I imagine that's not the only database not sticking with snake_case. Sooo....

I'd still like to use relations without defining the foreign keys explicitly. Like:
```
function foo(): HasOne
{
    $this->hasOne(Foo::class);
}
```
as it is possible by sticking to snake_casing.

`Illuminate\Database\Eloquent\Concerns\HasRelationships` and `Illuminate\Database\Eloquent\Model` very explicitly use `Str::snake()`, so apparently there is no way of changing that behaviour.

And just to be clear, I don't mean attributes and `snakeAttributes = false`, I mean the actual column names in the actual database.
Is there something I overlooked?

If I haven't, is that worth proposing as a feature? I mean, I probably already read most parts of the code to be touched.

**Comment by henzeb** [comment]
You can define table name in the Model belonging to the table.
